packages/contracts/docs/osx/01-how-it-works/02-framework/02-plugin-management/02-plugin-setup/index.md

```diff
@@ -1,72 +1,78 @@
+![Aragon OSx Plugins](https://res.cloudinary.com/dacofvu8m/image/upload/v1683225098/Screen_Shot_2023-05-04_at_14.31.25_r0qqut.png)
 
-Except for the gas costs required, the installation, update, and uninstallation of plugins is completely free.
+How this works:
 
-### Security Considerations
+- Although a Plugin is composed `Plugin` and `PluginSetup` contracts, the Aragon OSx protocol only knows of the `PluginSetup` contract.
+- Since the `PluginSetup` contract is the one in charge of installing the plugin into a DAO, it is the one also in charge of deploying the Plugin instace specific to that DAO with the parameters set by the DAO in advance. You can review how to build a `PluginSetup` contract [here](../../../../02-how-to-guides/02-plugin-development/index.md).
+- The `PluginSetup` contract then interacts with the Aragon OSx framework so that installing, updating, and uninstalling a plugin to a DAO becomes very simple for the end-user.
+- Publishing a Plugin into the Aragon OSx protocol is done through creating a plugin's first version. This generates a `PluginRepo` instance - registering all plugin versions. You can read more about that [here](../../../../02-how-to-guides/02-plugin-development/07-publication/index.md).
+- Except for the gas costs required, plugins are completely free.
 
-The plugin setup process is **security critical** because permissions are granted to third-party contracts.
-Safety was our top priority in the design and we wanted to make sure that the DAO knows exactly which contracts receive which permissions before processing and making sure that the `PluginSetup` contracts developed by third parties don’t obtain elevated permissions (i.e., the `ROOT_PERMISSION_ID` permission) on the installing DAO during the setup process.
+### How does this work?
+
+The `PluginSetup` process is **security critical** because permissions are granted to third-party contracts.
 
-This is why we split the **plugin setup in two steps**:
+Safety was our top priority in the design and we wanted to make sure that the DAO knows exactly which contracts receive which permissions before processing and making sure that the `PluginSetup` contracts developed by third parties don’t obtain elevated permissions (i.e., the `ROOT_PERMISSION_ID` permission) on the installing DAO during the setup process.
 
-1. **Preparation**
-2. **Application**
+This is why we split the `PluginSetup` development in two steps:
 
-Each plugin will then require both to run through the DAO's `PluginSetupProcessor` contract, which is part of the Aragon OSx framework.
+1. **Preparation:** Defining and encoding the permissions needed to install, uninstall, or update a plugin
```

Comment:
- Deploying the plugin and the helpers
- Requesting permissions to be applied
- Storing the configuration of the prepared setup

It's an unprivileged install script where developers can do whatever they need.